internal/internal_update.go

```diff
@@ -289,12 +289,15 @@ func defaultUpdateHandler(
 		if !IsReplaying(ctx) {
 			// we don't execute update validation during replay so that
 			// validation routines can change across versions
+			defer getState(ctx).dispatcher.setIsReadOnly(false)
+			getState(ctx).dispatcher.setIsReadOnly(true)
 			if err := envInterceptor.inboundInterceptor.ValidateUpdate(ctx, &input); err != nil {
 				callbacks.Reject(err)
 				return
 			}
```

Comment:
```suggestion
			err := func() error {
				defer getState(ctx).dispatcher.setIsReadOnly(false)
				getState(ctx).dispatcher.setIsReadOnly(true)
				return envInterceptor.inboundInterceptor.ValidateUpdate(ctx, &input)
			}()
			if err != nil {
				callbacks.Reject(err)
				return
			}
```

Mentioned in Slack convo. I think this localizes the read-only a bit better